tests/ga/test_extension.py

```diff
@@ -1171,11 +1195,121 @@ def test_get_ext_handling_status(self, *args):
 
         os.path.exists = orig_state
 
+    @patch("azurelinuxagent.common.conf.get_lib_dir")
+    def test_collect_ext_status(self, mock_lib_dir, *args):
+        """
+        This test validates that collect_ext_status correctly picks up the status file (sample-status.json) and then
+        parses it correctly.
+        """
+        ext_handler_i, extension = self._setup_extension_for_validating_collect_ext_status(mock_lib_dir,
+                                                                                           "sample-status.json", *args)
+        ext_status = ext_handler_i.collect_ext_status(extension)
+
+        self.assertEqual(ext_status.code, 1)
```

Comment:
Let's add strings describing what the error is. Not only they document the code, but also help a lot when debugging test failures. When this test fails you would just see
> 2 != 1
For sample strings look at test_wire.py in #1800

Reply:
Fixed. Added better documentation to describe what the numbers mean.